src/Compilers/CSharp/Portable/Binder/LockBinder.cs

```diff
@@ -55,9 +57,91 @@ internal override BoundStatement BindLockStatementParts(BindingDiagnosticBag dia
                 hasErrors = true;
             }
 
+            if (exprType?.IsWellKnownTypeLock() == true &&
```

Comment:
>exprType?.IsWellKnownTypeLock() == true [](http://example.com/codeflow?start=16&length=39)

If this condition is true, should we really report an error in the `if` above for ```!exprType.IsReferenceType``` situation. We don't really intend to dispose the `Lock` instance itself.

Reply:
The spec hasn't changed in this regard I think - it's still an error to `lock` on a `struct`. I guess it could be allowed in this case, but since the BCL `Lock` won't be a struct, I would say it's unnecessary.

Reply:
The design explicitly says that the Lock type must be a `class` and the intention (and LDM decision) is to not support any other "patterns" for now. But I will note in alternatives of the speclet that we could lift this check if the Lock type weren't a `class` anymore (or when we implement the open pattern in the future).